File: spec/lua/test_cast_to_top.lua
local luaunit = require("luaunit")

require("cast_to_top")

TestCastToTop = {}

function TestCastToTop:test_cast_to_top()
    local r = CastToTop:from_file("src/fixed_struct.bin")

    luaunit.assertEquals(r.code, 0x50)
    luaunit.assertEquals(r.header.code, 0x41)
    luaunit.assertEquals(r.header_casted.code, 0x41)
end

File: spec/python/test_cast_to_top.py
import unittest

from cast_to_top import CastToTop

class TestCastToTop(unittest.TestCase):
    def test_cast_to_top(self):
        r = CastToTop.from_file("src/fixed_struct.bin")

        self.assertEqual(r.code, 0x50)
        self.assertEqual(r.header.code, 0x41)
        self.assertEqual(r.header_casted.code, 0x41)
